cmd/sidecars/network-passt-binding/server/server.go

```diff
@@ -153,3 +164,17 @@ func Serve(server *grpc.Server, socket net.Listener, shutdownChan <-chan struct{
 
 	waitForShutdown(server, errChan, shutdownChan)
 }
+
+func primaryPodIfaceNameFromVMI(vmi *vmschema.VirtualMachineInstance) (string, error) {
+	primaryNet := vmispec.LookupPodNetwork(vmi.Spec.Networks)
+	if primaryNet == nil {
+		return "", fmt.Errorf("primary network was not found")
+	}
+
+	ifaceStatus := vmispec.LookupInterfaceStatusByName(vmi.Status.Interfaces, primaryNet.Name)
+	if ifaceStatus == nil {
+		return "", fmt.Errorf("primary network interface status was not found")
+	}
+
+	return ifaceStatus.PodInterfaceName, nil
```

Comment:
I think we need the check interface name is not empty

Reply:
Having the check here would make debugging easier and if there is a bug it will fail earlier.
In case of an issue it should save going back and forward with libvirt errors.

Reply:
Done.